app/src/main/java/com/example/sharewhatyoucanproject/login/LoginActivity.kt

```diff
@@ -0,0 +1,110 @@
+package com.example.sharewhatyoucanproject.login
+
+import android.app.ProgressDialog
+import android.content.Intent
+import android.os.Bundle
+import android.provider.Settings
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.DrawerActivity
+import com.example.sharewhatyoucanproject.SelectUserActivity
+import com.example.sharewhatyoucanproject.databinding.ActivityLoginBinding
+import com.example.sharewhatyoucanproject.models.UserType
+import com.example.sharewhatyoucanproject.utils.editSharedPreferencesSelector
+import com.example.sharewhatyoucanproject.utils.showToast
+
+class LoginActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityLoginBinding
+    lateinit var loginViewModel: LoginViewModel
+    lateinit var progressDialog: ProgressDialog
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        progressDialog = ProgressDialog(this)
+        progressDialog.setTitle("Please Wait")
+        supportActionBar?.hide()
+
+        loginViewModel = ViewModelProvider(
+            this,
+            LoginViewModelFactory(),
+        )[LoginViewModel::class.java]
+
+        val type = intent.getIntExtra("type", 0)
+
+        val typevar1 = UserType.values()[type]
+
+        loginViewModel.type = typevar1
+
+        loginViewModel.checkResult.observe(this) { result ->
+            result?.let {
+                progressDialog.dismiss()
+                // this.showToast(result)
+                // TODO show toast
+                loginViewModel.checkResult.value = null
```

Comment:
Only the view model should be able to modify its live data. You could subscribe to distinct changes instead of nulling it out. https://developer.android.com/reference/androidx/lifecycle/Transformations